tests/test-vandv/src/main/java/mil/emp3/test/emp3vv/navItems/kml_service_test/KMLServiceTest.java

```diff
@@ -0,0 +1,319 @@
+package mil.emp3.test.emp3vv.navItems.kml_service_test;
+
+import android.app.Activity;
+import android.content.DialogInterface;
+import android.os.Environment;
+import android.os.Handler;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.ArrayAdapter;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import mil.emp3.api.events.KMLSEvent;
+import mil.emp3.api.exceptions.EMP_Exception;
+import mil.emp3.api.interfaces.IKMLS;
+import mil.emp3.api.interfaces.IMap;
+import mil.emp3.api.KMLS;
+import mil.emp3.api.interfaces.IMapService;
+import mil.emp3.api.listeners.IKMLSEventListener;
+import mil.emp3.test.emp3vv.common.Emp3TesterDialogBase;
+import mil.emp3.test.emp3vv.common.ExecuteTest;
+import mil.emp3.test.emp3vv.common.NavItemBase;
+import mil.emp3.test.emp3vv.common.StyleManager;
+import mil.emp3.test.emp3vv.dialogs.utils.ErrorDialog;
+
+public class KMLServiceTest extends NavItemBase implements KmlsSettingsDialog.IKmlsSettingsDialogListener {
+
+    private static String TAG = KMLServiceTest.class.getSimpleName();
+
+    private final StyleManager styleManager;
+
+    private IKMLSEventListener[] kmlsEventListener = new IKMLSEventListener[ExecuteTest.MAX_MAPS];
+
+    String[] sampleURLs = {
+            "https://github.com/downloads/brazzy/nikki/example.kmz",
+            "https://developers.google.com/kml/documentation/KML_Samples.kml"
```

Comment:
These specifically hard coded for usability. There is support in VandV to build a File URL from files in /sdcard/testFiles. Additionally one can provide a URL by choosing the "CutomURL" option.